### PR TITLE
User length name

### PR DESCRIPTION
**Pull request by Cheshiriks**
User length name

Снимок экрана 2023-08-03 в 23 56 32

### CODE REVIEW

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/UsersDetailsController.kt [hunk 1]
```diff
@@ -100,8 +100,13 @@ class UsersDetailsController(
      */
     @PostMapping("/save")
     @PreAuthorize("isAuthenticated()")
+    @Suppress("MagicNumber")
     fun saveUser(@RequestBody newUserInfo: UserInfo, authentication: Authentication): Mono<StringResponse> =
             Mono.just(newUserInfo)
+                .filter { newUserInfo.name.length <= 22 }
```
Comment (sanyavertolet) [inline save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/UsersDetailsController.kt:106]: **Must** be moved to `const val`

Or even reuse `isValidName`

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/inputform/InputForms.kt [hunk 2]
```diff
@@ -14,6 +14,10 @@ private const val EMAIL_PLACEHOLDER = "test@example.com"
 private const val NAME_TOOLTIP = "Allowed symbols: letters, digits, dots, hyphens and underscores." +
         "No dot, hyphen or underscore at the beginning and at the end of the line."
 
+private const val NAME_ORG_PROJECT_TOOLTIP = "Name must not be longer than 22 characters." +
```
Comment (sanyavertolet) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/inputform/InputForms.kt:17]: 22 should be extracted

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/CreateOrganizationView.kt [hunk 3]
```diff
@@ -148,7 +148,8 @@ class CreateOrganizationView : AbstractView<Props, OrganizationSaveViewState>()
                                             form = InputTypes.ORGANIZATION_NAME
                                             conflictMessage = state.conflictErrorMessage
                                             textValue = state.organizationDto.name
-                                            validInput = (state.organizationDto.name.isEmpty() || state.organizationDto.validateName()) && state.conflictErrorMessage == null
+                                            validInput = (state.organizationDto.name.isEmpty() || state.organizationDto.validateName() ||
+                                                    state.organizationDto.name.length <= 22) && state.conflictErrorMessage == null
```
Comment (sanyavertolet) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/CreateOrganizationView.kt:152]: Must be moved to const val

**Review by sanyavertolet — CHANGES_REQUESTED** [review]

**Review by akuleshov7 — APPROVED** [review]

**Comment by nulls** [comment]
do we need a separate PR for `val 22`?

**Comment by sanyavertolet** [comment]
> do we need a separate PR for `val 22`?

not sure about separate PR, but it definitely should be fixed

**Pull request merged**